### PR TITLE
Feature/add jalali add date

### LINKED ISSUE #8: addDate is based on time.Time not the jalali calendar
when you add months or years you won't get an expected result in Jalali calendar.

### PR DESCRIPTION
**Pull request by ARMy4D**
fixed #8

### CODE REVIEW

File: ptime.go [hunk 1]
```diff
@@ -380,6 +414,23 @@ func (t *Time) Set(year int, month Month, day, hour, min, sec, nsec int, loc *ti
...
+	// Normalize nsec, sec, min, hour, overflowing into day.
+	sec, nsec = norm(sec, nsec, 1e9)
+	min, sec = norm(min, sec, 60)
+	hour, min = norm(hour, min, 60)
+	day, hour = norm(day, hour, 24)
+
+	// Normalize month, overflowing into year.
+	m := int(month) - 1
+	year, m = norm(year, m, 12)
+	if divider(25*year+11, 33) < 8 {
```
Comment (yaa110) [inline ptime.go]: you can use `IsLeap()` method to check for leap years to avoid repetition of the same calculation

File: ptime.go [hunk 2]
```diff
@@ -369,6 +369,40 @@ func (t *Time) SetUnix(sec, nsec int64, loc *time.Location) {
...
+		hi += n
+		lo -= n * base
+	}
+	return hi, lo
+}
+
+// norm returns nhi, nlo such that
+//	hi * base + lo == nhi * base + nlo
+//	0 <= nlo < base
+func normDay(hi, lo, base int) (nhi, nlo int) {
```
Comment (yaa110) [inline ptime.go]: don't use named return types

File: ptime.go [hunk 3]
```diff
@@ -369,6 +369,40 @@ func (t *Time) SetUnix(sec, nsec int64, loc *time.Location) {
 	t.SetTime(time.Unix(sec, nsec).In(loc))
 }
 
+// norm returns nhi, nlo such that
+//	hi * base + lo == nhi * base + nlo
+//	0 <= nlo < base
+func norm(hi, lo, base int) (nhi, nlo int) {
```
Comment (yaa110) [inline ptime.go]: don't use name return types

Comment (ARMy4D) [reply ptime.go]: its copied from time package exactly as it is

**Comment by ARMy4D** [comment]
@yaa110 can you make a release for this version?

**Pull request merged**